### PR TITLE
Added recommend resolutions for the "easier to fix" situations

### PR DESCRIPTION
**Pull request by jackchallen**
TA Tool is designed to highlight problems. It would be easier for customers/TSEs if we could place examples of the commands to rectify known issues, or at least how they would start fixing situations.

### CODE REVIEW

**Comment by jackchallen** [comment]
This one is pretty huge, I've been through most of our TA Tool checks adding recommended resolutions.

File: scripts.d/ta/510_check_for_noprefixroute.sh [hunk 1]
```diff
@@ -28,7 +28,10 @@ if [[ "${NOPREFIXROUTE_COUNT}" != "0" ]]; then
     echo "Certain IP addresses are configured with noprefixroute. This will inhibit the ability"
     echo "of certain cluster floating ips to accurately determine which link should be preferred"
     echo "The command \"ip -o -f inet route list match xxx.xxx.xxx.xxx/32 scope link\" needs to"
-    echo "Be able to return a device for each floating IP configured"
+    echo "be able to return a device for each floating IP configured"
+    echo "Recommended Resolution: remove the noprefixroute flag or otherwise ensure the IP"
```
Comment (vrragosta) [inline scripts.d/ta/510_check_for_noprefixroute.sh:32]: nix IP at the end of the sentence.

File: scripts.d/ta/520_bucket_and_process_uptime.sh [hunk 2]
```diff
@@ -36,12 +36,20 @@ CURRENT_TIME_EPOCH=$(                 date +%s)
 if [[ $((${CURRENT_TIME_EPOCH}-${MOST_RECENT_BUCKET_STARTTIME_EPOCH})) -lt 3600 ]]; then
     RETURN_CODE="254"
     echo "Weka buckets have been restarted within the last hour, or have never started. This may not be a problem on a new cluster"
-    echo "but could be indicative of problems (e.g. network flapping"
+    echo "but could be indicative of problems (e.g. network flapping)"
+    echo "Recommended Resolutions:"
+    echo " . If this is a new cluster,  or hosts have been upgraded/reboot, this is likely expected"
```
Comment (vrragosta) [inline scripts.d/ta/520_bucket_and_process_uptime.sh:41]: Remove spurious space after first comma.

Change upgraded/reboot to upgraded/rebooted.

File: scripts.d/ta/520_bucket_and_process_uptime.sh [hunk 3]
```diff
@@ -36,12 +36,20 @@ CURRENT_TIME_EPOCH=$(                 date +%s)
...
+    echo " . Otherwise the most likely cause is network problems, such as link flapping or congestion."
+    echo " . Review hardware and network stability, then contact customer success"
 fi
 if [[ $((${CURRENT_TIME_EPOCH}-${MOST_RECENT_PROCESS_STARTTIME_EPOCH})) -lt 3600 ]]; then
     RETURN_CODE="254"
     echo "Weka processes have been restarted within the last hour, or have never started. This may not be a problem on a new cluster"
     echo "but could be indicative of problems (e.g. network flapping"
+    echo "Recommended Resolutions:"
+    echo " . If this is a new cluster,  or hosts have been upgraded/reboot, this is likely expected"
```
Comment (vrragosta) [inline scripts.d/ta/520_bucket_and_process_uptime.sh:50]: Remove spurious space after first comma.

Change upgraded/reboot to upgraded/rebooted.

File: scripts.d/ta/755_wekapp424920_smbw_mask.sh [hunk 4]
```diff
@@ -34,11 +34,15 @@ if weka smb cluster | awk '/Type:/ && /smbw/' &> /dev/null; then
 
             if [[ $NUM_SHARES -ne $NUM_FILE_MASKS ]]; then
                 echo "WARN: there are $NUM_SHARES smbw shares, but only $NUM_FILE_MASKS shares with force_create_mode"
+                echo "Recommended Resolution: for each share, delete and re-create it to ensure this mode is set."
+                echo " NB: this will likely be service-affecting"
```
Comment (vrragosta) [inline scripts.d/ta/755_wekapp424920_smbw_mask.sh:38]: Change NB: to WARNING:

File: scripts.d/ta/755_wekapp424920_smbw_mask.sh [hunk 5]
```diff
@@ -34,11 +34,15 @@ if weka smb cluster | awk '/Type:/ && /smbw/' &> /dev/null; then
...
                 RETURN_CODE=254
             fi
 
             if [[ $NUM_SHARES -ne $NUM_DIR_MASKS ]]; then
                 echo "WARN: there are $NUM_SHARES smbw shares, but only $NUM_DIR_MASKS shares with force_directory_mode"
+                echo "Recommended Resolution: for each share, delete and re-create it to ensure this mode is set."
+                echo " NB: this will likely be service-affecting"
```
Comment (vrragosta) [inline scripts.d/ta/755_wekapp424920_smbw_mask.sh:45]: Change NB: to WARNING:

File: scripts.d/ta/785_asymmetric_mtu.sh [hunk 6]
```diff
@@ -15,6 +15,10 @@ for INDIVIDUAL_DRIVE_PROCESS in $(weka cluster process --backends --filter role=
     if [[ $(weka debug net peers --no-header ${INDIVIDUAL_DRIVE_PROCESS} --output inMTU,outMTU  | awk '{if($1 != $2) {print "yes"}}') == "yes" ]]; then
         host=$(weka cluster process ${INDIVIDUAL_DRIVE_PROCESS} --no-header -o hostname)
         echo "WARN: Asymmetric MTU detected for at least one peer of ${host}, process id ${INDIVIDUAL_DRIVE_PROCESS}"
+        echo "Recommended Resolution: The usual cause for this is assymetric routing, with different MTUs configured"
+        echo "along the two different paths. Run a tracepath/traceroute from each end of the backend<->client"
+        echo "connection, and determine if routes take different paths. It's likely that different will have"
```
Comment (vrragosta) [inline scripts.d/ta/785_asymmetric_mtu.sh:20]: different _**paths**_ will have

File: scripts.d/ta/795_netmask_mismatch.sh [hunk 7]
```diff
@@ -32,6 +32,11 @@ done
 
 if [[ ${RETURN_CODE} -eq 0 ]]; then
     echo "All Weka containers have consistent netmasks"
+else 
+    echo "Recommended Resolution: determine which of these netmasks is correct, and rectify the one with"
+    echo "the wrong configuration. If Weka needs re-configuring, this will be done with commands like"
+    echo " weka local resources --container <WEKA-CONTAINER> net remove <NIC>"
+    echo " weka local resources --container <WEKA-CONTAINER> net add    <NIC> --ips <IP-SPEC> --netmask <BIT-LENGTH>"
```
Comment (vrragosta) [inline scripts.d/ta/795_netmask_mismatch.sh:39]: Should we remove `--ips <IP-SPEC>`? It's more unusual than not to have to specify an IP here.

File: scripts.d/ta/810_use_only_readcache_for_protocols.sh [hunk 8]
```diff
@@ -14,6 +14,15 @@ for WEKA_CONTAINER in $(sudo weka local ps --output name --no-header | grep -w -
     if [[ ${MOUNTS_USING_WRITECACHE} != "0" ]]; then
         echo "WARN: container ${WEKA_CONTAINER} - used for protocols - is using writecache on host ${HOSTNAME}"
         echo "Refer to ${JIRA_REFERENCE} for more details"
+        if [[ ${WEKA_CONTAINER} =~ "s3" ]]; then
+            echo "Recommended Resolution: for s3, use the following (brief service interruption):"
+            echo " weka s3 cluster update --mount-options readcache -f"
+        elif [[ ${WEKA_CONTAINER} =~ "smb" ]]; then
+            echo "Recommended Resolution: for smb, for each share, delete it and re-add it (service interruption)"
```
Comment (vrragosta) [inline scripts.d/ta/810_use_only_readcache_for_protocols.sh:21]: Does this resolution require a minimum Weka version?

File: scripts.d/ta/810_use_only_readcache_for_protocols.sh [hunk 9]
```diff
@@ -14,6 +14,15 @@ for WEKA_CONTAINER in $(sudo weka local ps --output name --no-header | grep -w -
...
+        elif [[ ${WEKA_CONTAINER} =~ "ganesha" ]]; then
+            echo "Recommended Resolution: for NFS, for each share, delete it and re-add it (service interruption)"
```
Comment (vrragosta) [inline scripts.d/ta/810_use_only_readcache_for_protocols.sh:23]: Does this resolution require a minimum Weka version?

File: scripts.d/ta/430_nvme_used_capacity_vs_maximum.sh [hunk 10]
```diff
@@ -27,6 +27,13 @@ echo ${WEKA_SSD_USED_BYTES}
 
 # if we've allocated more than half the maximum theoretical SSD space, warn
 if [[ $((${WEKA_SSD_USED_BYTES}*2)) -gt ${WEKA_THEORETICAL_MAX_SSD_BYTES} ]] ; then 
+    echo "You have used a significant proporation of the theoretical maximum"
```
Comment (vrragosta) [inline scripts.d/ta/430_nvme_used_capacity_vs_maximum.sh:30]: **_proportion_**

File: scripts.d/ta/440_hostnames_rfc952.sh [hunk 11]
```diff
@@ -23,6 +23,7 @@ GREP_RESULT=$(echo ${SHORT_HOSTNAME} | grep "[^-a-z0-9.]")
 if [[ $? -eq 0 ]]; then
     echo "The hostname ${SHORT_HOSTNAME} appears to contain a character other than [a-z], -, and [0-9]."
     echo "Refer to RFC 952 for more information"
+    echo "Recommended resolution: change the hostname to include only alphanumerics and underscore"
```
Comment (vrragosta) [inline scripts.d/ta/440_hostnames_rfc952.sh:26]: **_underscores_**

File: scripts.d/ta/490_ip_route_metrics.sh [hunk 12]
```diff
@@ -46,6 +46,8 @@ if [[ ${NUMBER_OF_OVERLAPPING_ROUTES_WITH_METRICS} -gt "1" ]]; then
     echo "that these entries will negatively affect the performance of e.g. floating IP"
     echo "addresses. In any case it is unlikely that preferential IP routes are of"
     echo "benefit in a high-performance local network"
+    echo "Recommended Resolution: review the output of \"ip route\" and rationalise the routes,"
```
Comment (vrragosta) [inline scripts.d/ta/490_ip_route_metrics.sh:49]: **_rationalize_**

File: scripts.d/ta/530_high_drive_read_ssd_ratio.sh [hunk 13]
```diff
@@ -34,6 +34,9 @@ if [[ ${HIGHER_THAN_EXPECTED} == "YES" ]]; then
     echo "The ratio of NVMe read requests vs DRIVE node read operations is higher than expected over the last ${TIME_TO_EXAMINE}"
     echo "This could indicate a number of things, such as splitting of read requests or perhaps read amplification"
     echo "Review ${JIRA_REFERENCE} for details"
+    echo "Recommended Resolutions:"
+    echo " . This may be expected behaviour for your workload"
```
Comment (vrragosta) [inline scripts.d/ta/530_high_drive_read_ssd_ratio.sh:38]: **_behavior_**

**Review by vrragosta — CHANGES_REQUESTED** [review]
Minor changes requested -- i.e. spelling, etc.

File: scripts.d/ta/810_use_only_readcache_for_protocols.sh (hunk 8 above)
Comment (jackchallen) [reply scripts.d/ta/810_use_only_readcache_for_protocols.sh:21]: I'm going to ignore this question. If this happens in real life and we do need an update I think re-running the TA Tool is enough to highlight nothing changed, and we'll deal with as SUpport case

File: scripts.d/ta/810_use_only_readcache_for_protocols.sh (hunk 9 above)
Comment (jackchallen) [reply scripts.d/ta/810_use_only_readcache_for_protocols.sh:23]: I'm going to ignore this question. If this happens in real life and we do need an update I think re-running the TA Tool is enough to highlight nothing changed, and we'll deal with as SUpport case

**Review by vrragosta — APPROVED** [review]
A-OK

**Pull request merged**